docs/pages/modules/mocking.mdx

```diff
@@ -0,0 +1,56 @@
+---
+title: 'Mocking native calls in Expo modules'
+sidebar_title: Mocking native calls
+description: Learn about mocking native calls in Expo modules.
+---
+
+import { BoxLink } from '~/ui/components/BoxLink';
+import { Grid01Icon } from '@expo/styleguide-icons';
+import { Terminal } from '~/ui/components/Snippet';
+
+The recommended way to write unit tests for an Expo project is to use [Jest](https://jestjs.io/) and the `jest-expo` preset.
+
+To write a unit test for an app that uses native code, you need to mock native calls.
+The term **Mocking** means to replace the actual implementation of a function with a fake version that does not perform any actions. This approach is useful for running unit tests on a local computer as it involves bypassing the need for native code which is only available on a device, and any code that calls native functions on a local machine will not work.
+
+Expo SDK includes a set of default mocks for each of our community packages. You can also mock any JS code yourself using built-in Jest APIs such as [mock functions](https://jestjs.io/docs/mock-functions).
+
+However, to provide default mocks in your Expo Module, we offer a method to bundle them. This ensures that when your module user runs unit tests, they will automatically use a mocked implementation.
+
+## Providing mocks for a module
+
+Create a file with the same name as the native module you want to mock and place it in your module's **mocks** directory. Make sure to export the mock implementation from this file.
+The `jest-expo` preset will automatically return the exported functions as a result of a `requireNativeModule` call when running during a unit test.
```

Comment:
```suggestion
The `jest-expo` preset will automatically return the exported functions because of a `requireNativeModule` call when running during a unit test.
```
Suggestion to fix vale warning:

![CleanShot 2024-02-12 at 15 06 47@2x](https://github.com/expo/expo/assets/10234615/dad12ec0-270f-40c7-b505-71ffe85f8c76)